ldj-readme-renderer/render.js

```diff
@@ -8,45 +8,43 @@
-[![Technical Debt](https://sonarcloud.io/api/project_badges/measure?project=Ali-chakaroun_ISEP-LivingDocumentation&metric=sqale_index)](https://sonarcloud.io/summary/new_code?id=Ali-chakaroun_ISEP-LivingDocumentation)
-[![Reliability Rating](https://sonarcloud.io/api/project_badges/measure?project=Ali-chakaroun_ISEP-LivingDocumentation&metric=reliability_rating)](https://sonarcloud.io/summary/new_code?id=Ali-chakaroun_ISEP-LivingDocumentation)
-[![Maintainability Rating](https://sonarcloud.io/api/project_badges/measure?project=Ali-chakaroun_ISEP-LivingDocumentation&metric=sqale_rating)](https://sonarcloud.io/summary/new_code?id=Ali-chakaroun_ISEP-LivingDocumentation)
-# Living Documentation Java
-
-This project is an extension of the [Living Documentation][ldoc] set of tools to
-Java. As part of the Living Documentation toolchain, it can be used to generate
-documentation and diagrams from source code so that documentation is always
-up-to-date.
-
-## Analyzer
-
-The analyzer parses Java projects into Living Documentation JSON files that can
-be rendered by Living Documentation renderers. It accepts the following command
-line options:
+const json = JSON.parse(fs.readFileSync(process.argv[2]));
 
+// Section for rendering the command line options
+const CMD_LINE_TABLE_TEMPLATE = `
 | Short option | Long option | Description |
 | ------------ | ----------- | ----------- |
 ~
-
-[ldoc]: https://github.com/eNeRGy164/LivingDocumentation
 `;
 
-function render(d) {
+function renderCmdOptions(d) {
   if (d.Name === 'addOption' || d.Name === 'addRequiredOption') {
     const args = d.Arguments.map(a => a.Text).map(JSON.parse);
     const short = args[0] ? `\`-${args[0]}\`` : '';
     const long = args[1] ? `\`--${args[1]}\`` : '';
     const required = d.Name.includes('Required') ? '(Required.)' : '';
     return `| ${short} | ${long} | ${args[3]} ${required} |\n`;
   } else if (d instanceof Array) {
-    return d.map(render).join('');
+    return d.map(renderCmdOptions).join('');
   } else if (d instanceof Object) {
-    return Object.values(d).map(render).join('');
+    return Object.values(d).map(renderCmdOptions).join('');
   }
 }
 
-const json = JSON.parse(fs.readFileSync(process.argv[2]));
-fs.writeFileSync(OUTPUT, TEMPLATE.replace('~', () => render(json)));
+const cmdOptionsTable = CMD_LINE_TABLE_TEMPLATE.replace('~', () => renderCmdOptions(json));
+const visitor = json.filter((type) => type.FullName === 'com.infosupport.ldoc.analyzerj.AnalysisVisitor')[0];
+
+// Combining everything
+try {
+  const readmeTemplate = fs.readFileSync(README_TEMPLATE, 'utf8');
+  const finalReadme = readmeTemplate.replace('[~cmd-line-options]', cmdOptionsTable);
+  fs.writeFileSync(OUTPUT, finalReadme);
+} catch (err) {
+  console.error(err);
```

Comment:
I think the default behavior for an uncaught exception is to write the error to the console - does catching and logging it explicitly here make an important difference?

If not, I'd suggest removing the try/catch. That also makes it consistent with the `readFileSync` of the JSON file, above.

Reply:
For the purpose of this renderer does indeed not make a difference. Agreed, so adjusted